jmclient/jmclient/blockchaininterface.py

```diff
@@ -406,6 +407,33 @@ def get_block_time(self, blockhash):
         except JsonRpcError:
             return self.rpc('getblock', [blockhash])['time']
 
+    def get_tx_merkle_branch(self, txid, blockhash=None):
+        """
+        Will fail and return None if the containing block has been pruned
```

Comment:
There's an argument for following a paradigm that I've started to use in a few other places: you have a function that you want to return a thing, but sometimes is not able to, for various reasons. Make it return a tuple (thing, errormsg) where errormgs is None when thing is found successfully. This way callers can report the reason for the failure when it occurs.
(This function seems like it can fail for 2 or 3 distinct reasons).

Reply:
Thinking about it, I think I'd prefer raising some kind of exception, after all the original reason for exceptions was to make code more readable by separating error handling from normal code. So I'll recode this to use ValueError